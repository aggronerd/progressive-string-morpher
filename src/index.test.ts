import {calculateTransitionSteps} from "./index";

let targetString: string;
let initialString: string;

describe('calculateTransitionSteps', () => {
    const subject = () => calculateTransitionSteps(initialString, targetString);

    describe('Both empty strings', () => {
        beforeEach(() => {
            initialString = '';
            targetString = '';
        });

        it('Returns expected', () => {
            expect(subject()).toEqual(['']);
        })
    });

    describe('Empty to something', () => {
        beforeEach(() => {
            initialString = '';
            targetString = 'Something';
        });

        it('Returns expected', () => {
            expect(subject()).toEqual([
                '',
                'S',
                'So',
                'Som',
                'Some',
                'Somet',
                'Someth',
                'Somethi',
                'Somethin',
                'Something',
            ]);
        })
    });

    describe('Something to empty', () => {
        beforeEach(() => {
            initialString = 'Something';
            targetString = '';
        });

        it('Returns expected', () => {
            expect(subject()).toEqual([
                'Something',
                'something',
                'omething',
                'mething',
                'ething',
                'thing',
                'hing',
                'ing',
                'ng',
                'g',
                '',
            ]);
        })
    });

    describe('Not an anagram!', () => {
        beforeEach(() => {
            initialString = 'This is not an anagram';
            targetString = 'It certainly is not!'
        });

        it('returns expected', () => {
            expect(subject()).toMatchSnapshot()
        });
    });

    describe('Aggronerd!!!', () => {
        beforeEach(() => {
            initialString = 'Greg Doran';
            targetString = 'Aggronerd!!!';
        });

        it('returns expected', () => {
            expect(subject()).toEqual([
                'Greg Doran',
                'greg Doran',
                'greg doran',
                'grge doran',
                'grgedoran',
                'grgeodran',
                'grgeordan',
                'grgeoradn',
                'grgeorand',
                'ggreorand',
                'ggroerand',
                'ggroearnd',
                'ggroeanrd',
                'ggroaenrd',
                'ggroanerd',
                'ggraonerd',
                'ggaronerd',
                'gagronerd',
                'aggronerd',
                'aggronerd!',
                'aggronerd!!',
                'aggronerd!!!',
                'Aggronerd!!!',
            ])
        });
    });

    describe('He bugs gore!', () => {
        beforeEach(() => {
            targetString = 'He bugs Gore!';
            initialString = 'George Bush';
        });

        it('returns expected', () => {

            expect(subject()).toEqual([
                "George Bush",
                "george Bush",
                "george bush",
                "egorge bush",
                "egogre bush",
                "egogr ebush",
                "egogr beush",
                "egogr buesh",
                "egogr buseh",
                "egogr bushe",
                "eggor bushe",
                "eggo rbushe",
                "eggo brushe",
                "eggo burshe",
                "eggo busrhe",
                "eggo bushre",
                "egg obushre",
                "egg boushre",
                "egg buoshre",
                "egg busohre",
                "egg bushore",
                "eg gbushore",
                "eg bgushore",
                "eg bugshore",
                "eg busghore",
                "eg bushgore",
                "e gbushgore",
                "e bgushgore",
                "e bugshgore",
                "e bughsgore",
                "e buhgsgore",
                "e bhugsgore",
                "e hbugsgore",
                "eh bugsgore",
                "he bugsgore",
                "he bugs gore",
                "he bugs gore!",
                "He bugs gore!",
                "He bugs Gore!"
            ])
        });
    });
});
