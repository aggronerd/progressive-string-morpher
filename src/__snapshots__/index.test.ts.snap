// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calculateAll Not an anagram! returns expected 1`] = `
Array [
  "This is not an anagram",
  "this is not an anagram",
  "tis is not an anagram",
  "its is not an anagram",
  "it sis not an anagram",
  "it iss not an anagram",
  "it is not an anagram",
  "it i snot an anagram",
  "it i nsot an anagram",
  "it i nsto an anagram",
  "it i nst oan anagram",
  "it i nst aon anagram",
  "it i nst ano anagram",
  "it i nst anoanagram",
  "it i nst anonagram",
  "it i nst anoagram",
  "it i nst anogram",
  "it i nst anoram",
  "it i nst anroam",
  "it i nst anrom",
  "it i nst anro",
  "it in st anro",
  "it in ts anro",
  "it in tsa nro",
  "it in tsa rno",
  "it int sa rno",
  "it int as rno",
  "it int asr no",
  "it itn asr no",
  "it itna sr no",
  "it itna rs no",
  "it tina rs no",
  "it tian rs no",
  "it tianr s no",
  "it tainr s no",
  "it tairn s no",
  "it tarin s no",
  "it train s no",
  "it rtain s no",
  "it crtain s no",
  "it certain s no",
  "it certainl s no",
  "it certainly s no",
  "it certainly is no",
  "it certainly is not",
  "it certainly is not!",
  "It certainly is not!",
]
`;

exports[`calculateAll Rome example returns expected 1`] = `
Array [
  "More",
  "more",
  "omre",
  "orme",
  "rome",
  "rome!",
  "Rome!",
]
`;
